src/NRediSearch/Client.cs

```diff
@@ -55,45 +54,36 @@ public ConfiguredIndexOptions(IndexOptions options)
             /// Set a custom stopword list.
             /// </summary>
             /// <param name="stopwords">The new stopwords to use.</param>
-            public ConfiguredIndexOptions SetStopwords(params string[] stopwords)
+            public IndexOptions SetStopwords(params string[] stopwords)
             {
-                _stopwords = stopwords ?? throw new ArgumentNullException(nameof(stopwords));
-                if (stopwords.Length == 0) _options |= IndexOptions.DisableStopWords;
-                else _options &= ~IndexOptions.DisableStopWords;
+                _stopwords = stopwords;
+
                 return this;
             }
 
             internal void SerializeRedisArgs(List<object> args)
             {
-                SerializeRedisArgs(_options, args);
-                if (_stopwords != null && _stopwords.Length != 0)
-                {
-                    // note that DisableStopWords will not be set in this case
-                    args.Add("STOPWORDS".Literal());
-                    args.Add(_stopwords.Length.Boxed());
-                    foreach (var word in _stopwords)
-                        args.Add(word);
-                }
-            }
-
-            internal static void SerializeRedisArgs(IndexOptions flags, List<object> args)
-            {
-                if ((flags & IndexOptions.UseTermOffsets) == 0)
+                if (!_options.HasFlag(IndexOption.UseTermOffsets))
                 {
                     args.Add("NOOFFSETS".Literal());
                 }
-                if ((flags & IndexOptions.KeepFieldFlags) == 0)
+                if (!_options.HasFlag(IndexOption.KeepFieldFlags))
```

Comment:
note that `Enum.HasFlag<T>` is a boxing operation; perhaps add a helper method:

``` c#
private bool HasOption(IndexOption option) => (_options & option) != 0;
```

then

``` c#
if (HasOption(IndexOption.KeepFieldFlags)) {...}
```

etc

Reply:
I reverted this logic back to how it was before.